include/drivers/spi.h

```diff
@@ -517,6 +557,24 @@ static inline int spi_write(const struct device *dev,
 	return spi_transceive(dev, config, tx_bufs, NULL);
 }
 
+/**
+ * @brief Write data to a SPI bus specified in @p spi_dt_spec.
+ *
+ * This is equivalent to:
+ *
+ *     spi_write(spec->bus, &spec->config, tx_bufs);
+ *
+ * @param spec SPI specification from devicetree
+ * @param tx_bufs Buffer array where data to be sent originates from.
+ *
+ * @retval a value from spi_write()
+ */
+static inline int spi_write_dt(const struct spi_dt_spec *spec,
```

Comment:
Do we expect to have `_dt` API helpers for other drivers as well? That is, should a developer expect to find, e.g., `gpio_pin_get_dt`? I think this is an important point for APIs consistency. In the case of GPIO, `_dt` API helpers are only available for configuration related APIs, but nothing else.

I don't have a strong opinion here, but I'd personally not mix DT with _operation_ APIs. Non-blocking comment, but I think it's worth discussing before merging these changes.

Reply:
> Do we expect to have _dt API helpers for other drivers as well?

I certainly hope so, writing boilerplate when you don't need to is annoying.
#36477 added the GPIO versions you mentioned :)